main/coursecopy/recycle_course.php

```diff
@@ -55,14 +59,41 @@
         $course = $cb->build();
     }
     $recycle_type = '';
+    $fullDelete = 0;
+    $courseCodeConfirmation ='';
+    if (isset($_POST['course_code_confirmation']))
+        {
+            $courseCodeConfirmation = $_POST['course_code_confirmation'];
+        }
+
     if (isset($_POST['recycle_option']) && $_POST['recycle_option'] === 'full_backup') {
         $recycle_type = 'full_backup';
+        $fullDelete = 1;
+
     } elseif (isset($_POST['action']) && $_POST['action'] === 'course_select_form') {
         $recycle_type = 'select_items';
     }
+
     $cr = new CourseRecycler($course);
-    $cr->recycle($recycle_type);
-    echo Display::return_message(get_lang('RecycleFinished'), 'confirm');
+
+    if ($recycle_type == 'full_backup')
+        {
+            /* to delete, course code confirmation must be equal that current course code */
+            if ($current_course_code == $courseCodeConfirmation)
+                {
+                    $cr->recycle($recycle_type);
+                } else
+                {
+                    $messageFailCourseCode = '<p>' . get_lang('CourseRegistrationCodeIncorrect') . '</p>';
+                    $messageFailCourseCode .= '<p><a class="btn btn-primary" href="' . api_get_self() . '?' . api_get_cidreq() . '">' . get_lang('BackToPreviousPage') . '</a></p>';
```

Comment:
Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 180 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5721344)